### PR TITLE
Update to v1.64.0

### PR DESCRIPTION
**Pull request by marcoesters**
Update rust to the latest version. `cargo-license-bundle` needs an updated `rust` version to compile, so the we update to the latest version.

# Package info

Home: https://www.rust-lang.org
Development: https://github.com/rust-lang/rust
Documentation: https://doc.rust-lang.org/std/
Diff: https://github.com/rust-lang/rust/compare/1.56.0...1.64.0

### CODE REVIEW

**Review by varlackc — COMMENTED** [review]
The following changes might useful

license_url: https://github.com/rust-lang/rust/blob/master/COPYRIGHT

there is a redirect in the `doc_url` to the following:

doc_url: https://www.rust-lang.org/learn

**Review by ryanskeith — COMMENTED** [review]
These seem fine to me.

**Review by varlackc — APPROVED** [review]
LGTM

File: recipe/activate.sh [hunk 1]
```diff
@@ -21,7 +21,7 @@ echo "linker = \"${CONDA_PREFIX}/bin/x86_64-apple-darwin13.4.0-clang\"" >> ${CAR
 echo "[target.i686-unknown-linux-gnu]" >> ${CARGO_CONFIG}
 echo "linker = \"${CONDA_PREFIX}/bin/i686-conda_cos6-linux-gnu-cc\"" >> ${CARGO_CONFIG}
 echo "[target.x86_64-unknown-linux-gnu]" >> ${CARGO_CONFIG}
-echo "linker = \"${CONDA_PREFIX}/bin/x86_64-conda_cos6-linux-gnu-cc\"" >> ${CARGO_CONFIG}
+echo "linker = \"${CONDA_PREFIX}/bin/x86_64-conda_cos7-linux-gnu-cc\"" >> ${CARGO_CONFIG}
```
Comment (katietz) [inline recipe/activate.sh:24]: I think this change is not necessary, nor wise.  Old version of gcc where using conda_cos6 as vendor name.  By change it here, we make those all compilers incompatible.  New compilers are using actually conda as vendor-name (just on ppc64le we still should stick additionally to a conda_cos7 name ... caused by cdt)

File: recipe/meta.yaml [hunk 2]
```diff
@@ -41,16 +41,22 @@ outputs:
 about:
   home: https://www.rust-lang.org
   summary: A safe systems programming language (conda activation scripts)
-  license: BSD
+  license: BSD-3-Clause
   license_family: BSD
   license_file: LICENSE.txt
+  license_url: https://github.com/AnacondaRecipes/rust-activation-feedstock/blob/master/LICENSE.txt
```
Comment (katietz) [inline recipe/meta.yaml:47]: I really don't like those license_url tags, if there is already a license_file one

File: recipe/activate.sh (hunk 1 above)
Comment (marcoesters) [reply recipe/activate.sh]: The test fails when reverting to cos6: https://cloud.prefect.io/anaconda/flow-run/93ba490a-fe0b-4db8-8456-7c680a32bfaa?logs=&logId=46723c42-175d-4f4d-954b-562e2d9848db

**Review by katietz — APPROVED** [review]
thanks

**Pull request merged**